modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -259,18 +259,44 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
         if groupby_args["level"] is not None:
             raise NotImplementedError("levels are not supported for groupby")
 
-        groupby_cols = by_frame.columns.tolist()
-        agg_cols = [col for col in self.columns if col not in by_frame.columns]
+        groupby_cols = by_frame.columns
+        if isinstance(agg, dict):
+            agg_cols = agg.keys()
+        elif len(self.columns) == 1:
+            # This branch is trying to detect grouping on Series object, in that case,
+            # `by_cols == agg_cols`, so we have to explicitly set proper `agg_cols`
+            # to avoid its filtering into an empty list that happens for DataFrame cases.
+            # HACK: This check is naive and will be also true for single-column frames,
+            # however there are no more options to distinguish between Series and DataFrame
+            # at this level of abstraction. So potentially we're having a mismatch
+            # from pandas at grouping on single-column frames.
+            agg_cols = self.columns
+        else:
+            agg_cols = [col for col in self.columns if col not in by_frame.columns]
 
         # Create new base where all required columns are computed. We don't allow
         # complex expressions to be a group key or an aggeregate operand.
         assert isinstance(by_frame._op, TransformNode), "unexpected by_frame"
-        exprs = OrderedDict(((col, by_frame.ref(col)) for col in groupby_cols))
+
+        def generate_by_name(by):
+            """
+            Generates unuqie name for 'by' column in the resulted frame
+            to avoid names intersections with aggregated columns.
+            """
+            if groupby_args["as_index"]:
+                return f"__index__0_{by}"
+            elif by in agg_cols:
+                return f"by_{by}"
+            else:
+                return by
+
+        exprs = OrderedDict(
+            ((generate_by_name(col), by_frame.ref(col)) for col in groupby_cols)
+        )
+        groupby_cols = list(exprs.keys())
         exprs.update(((col, self.ref(col)) for col in agg_cols))
         exprs = translate_exprs_to_base(exprs, base)
-        base_cols = Index.__new__(
-            Index, data=list(exprs.keys()), dtype=self.columns.dtype
-        )
+        base_cols = Index.__new__(Index, data=exprs.keys(), dtype=self.columns.dtype)
```

Comment:
```suggestion
        base_cols = Index.__new__(Index, data=groupby_cols, dtype=self.columns.dtype)
```
?

Reply:
`groupby_cols` isn't equal to the `exprs.keys()` since the `expr` dictionary is modified right after `groupby_cols` assignment